vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -475,7 +475,7 @@ else if (!decksToDelete.isEmpty()) {
           final Board board = drawPile.getConfigureBoard(true);
           if ((boardName == null) || boards.contains(board)) {
             for (final DrawPile drawPile2 : foundDrawPiles) {
-              if (drawPile.getAttributeValueString(SetupStack.NAME).equals(drawPile2.getAttributeValueString(SetupStack.NAME))) {
+              if (Objects.equals(drawPile.getAttributeValueString(SetupStack.NAME), drawPile2.getAttributeValueString(SetupStack.NAME))) {
```

Comment:
Will this introduce a new problem if both `drawPile.getAttributeValueString(SetupStack.NAME)` and `drawPile2.getAttributeValueString(SetupStack.NAME)` are `null`? In that case, `Object.equals()` returns `true`, so the two piles will register as matching.